### ISSUE TITLE
Limited vectorization with mixed input types

### ISSUE THREAD

**Issue by naoyam** [csarofeen/pytorch#2466]
When a fusion has inputs with different types, the maximum vectorization factor is bounded by the largest data type. For example, a fp16 batchnorm with fp32 weights is vectorized by 4 at maximum as that's the limit for fp32.
